### PR TITLE
mark iv field obsolete

### PR DESCRIPTION
**Pull request by jsvd**
Thanks for contributing to Logstash! If you haven't already signed our CLA, here's a handy link: https://www.elastic.co/contributor-agreement/

### CODE REVIEW

File: lib/logstash/filters/cipher.rb [hunk 1]
```diff
@@ -147,32 +130,22 @@ def filter(event)
     #If decrypt or encrypt fails, we keep it it intact.
     begin
 
-      if (event[@source].nil? || event[@source].empty?)
```
Comment (untergeek) [inline lib/logstash/filters/cipher.rb:150]: Wow.  Were we really this out of date?

Comment (jsvd) [reply lib/logstash/filters/cipher.rb:150]: yes :(

**Comment by untergeek** [comment]
This one's not passing the tests, for _any_ of the builds.  Did we miss something here?

**Comment by untergeek** [comment]
For example:
```
  1) LogStash::Filters::Cipher single event, encrypt/decrypt aes-128-cbc, 16b RANDOM IV, 16b key, b64 encode validate decrypted message
     Failure/Error: expect(result.get("message_decrypted")).to eq(result.get("message"))
     
       expected: "شسيبشن٤٤ت٥ت داھدساققبمر фывапролдзщшгнекутцйячсмить asdfghjklqpoiuztreyxcvbnm,.-öäü+ä123ß´yö.,;LÖÜ*O 來少精清皆人址法田手扌打表氵開日大木裝 1234567890\#$%^&*()!@#;:'?.>,<testAESEn+=_-~`}]"
            got: nil
     
       (compared using ==)
     # ./spec/filters/cipher_spec.rb:81:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
```

**Comment by jsvd** [comment]
Yeah this hasn't passed for a long time, I'll take a second look and try to make it work again

**Comment by jsvd** [comment]
I just decided to refactor the specs to not use devutils and the insist framework, + fixed the bug this PR introduced and was making the specs fail

**Comment by jsvd** [comment]
I can't replicate the failures i see in travis, I ran the tests against logstash-core master and 6.x just fine..ideas?

**Comment by original-brownbear** [comment]
@jsvd LGTM as far as the code goes, I retriggered Travis now let's see if we get lucky and time was on our side here.
If it still busts for `master`, maybe it's still worth it merging this now and looking into it again in the next issue? At least `5.6` would be fixed ... better than what we have in `master`?

**Pull request merged**